### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by AbdelazizSoliman**
In this PR I implemented the following tasks:
 

- [ ] Created a method called current_user in your ApplicationController to make current user data available to all controllers.

- [ ]  Created forms to perform the following functions:

   1. Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
   2. Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

- [ ]  Allowed Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 <h2>Post <%= "#" + @post.id.to_s %> by <%= @user.name %></h2>
 <div class='post-interactions'>
-    <span>Comments: <%= @post.comments_counter %>,</span>
-    <span>Likes: <%= @post.likes_counter %></span>
+  <span>Comments: <%= @post.comments_counter %>,</span>
+  <span>Likes: <%= @post.likes_counter %></span>
 </div>
+<% if current_user.present? %>
+  <div class="post-actions">
+    <%= button_to "Like", like_user_post_path(@user, @post), method: :post, remote: true %>
```
Comment (Lucas-Masaba) [inline app/views/posts/show.html.erb:11]: - [ ] I get an error when I try to create a like with your form. Please make sure when a user clicks a like, it is incremented on the post.
image

File: app/views/comments/new.html.erb [hunk 2]
```diff
@@ -1,9 +1,17 @@
-<h1>New comment</h1>
+<a href="<%= user_post_path(@user, @post) %>">
+  <button id="back-btn">⬅</button>
+</a>
+<h1>Create a New Comment</h1>
+<h2>Post <%= @post.id %></h2>
+<h2>Current User <%= @user.name %></h2>
 
-<%= render "form", comment: @comment %>
+<%= form_with model: @comment, url: user_post_comments_path(@user, @post), local: true do |form| %>
```
Comment (Lucas-Masaba) [inline app/views/comments/new.html.erb:8]: - [ ] When I try to create a comment using this form, it is not added to the user. On clicking the submit button, nothing is added to the database and their is no redirection. Please fix this 
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @AbdelazizSoliman  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Well written commit messages :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/comments/new.html.erb (hunk 2 above)
Comment (AbdelazizSoliman) [reply app/views/comments/new.html.erb:8]: Still trying to fix.

Comment (AbdelazizSoliman) [reply app/views/comments/new.html.erb:8]: Solved!

**Review by mwanawabangona — APPROVED** [review]
Hi @AbdelazizSoliman,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbdelazizSoliman** [comment]
Thank you @mwanawabangona

**Pull request merged**